### PR TITLE
Budget app

### PR DESCRIPTION
**Pull request by AmineDerbal**
In this pull request ♻️ 

- Create a budget app that registers and log in and keeps track of all expenses in the categories 👍  
- Rename the entity table to expenses 👍 
- Create an intermediary table between expenses and groups to solve the many to many relationship 👍 
-  Secure app from n+1 problems by using through relationship in the model 👍 
- Create categories and expenses pages along with the forms to add new category and expense 👍 
- when adding an expense you can select either one or several categories 👍 
- The icons for the group table are saved in the assets/images folder 👍 
- Create units and integrations tests for the pages 👍 
- presentation video : https://www.loom.com/share/75541b5f79bc4c3ea7701da3b5185450?sid=d8124c46-4952-4244-9030-536f7e76e65b

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,195 @@
...
+
+- Use database migration files to maintain database schema.
+- Use Devise gem for authentication
+- Limit access to web app resources based on authorization rules
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [You can check the live demo here](https://budget-expense-app.onrender.com/)
```
Comment (elisha2kyakpo1) [inline README.md:78]: - [ ] I got an error when I tried to register on the app, kindly fix the error in production for the app to work as expected.

live

File: README.md [hunk 2]
```diff
@@ -1,24 +1,195 @@
...
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 📺 Presentation <a name="presentation"></a>
+- [Link to video presentation](https://www.loom.com/share/75541b5f79bc4c3ea7701da3b5185450?sid=18c2c3e0-870c-4904-8221-5c065014e947)
+
```
Comment (elisha2kyakpo1) [inline README.md]: - [ ] Great job recording a video as per requirement, unfortunately, the video doesn't have sound. Kindly record another video and follow all the video requirements.

video

File: app/views/groups/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+          <div class='group-amount'> $<%=number_with_precision(group.expenses.sum(:amount), precision: 2) || 0.00 %> </div>
+   
+      <% end %>
+    <% end %>
+  <% else %>
+    <h1 class='heading blue-text'>No category exists</h1>
+  <% end %>
+
+  <%= link_to "ADD CATEGORY", new_group_path, method: :get, class: "btn", id:'add-category' %>
+</div>
```
Comment (elisha2kyakpo1) [inline app/views/groups/index.html.erb]: - [ ] Kindly make sure that the design matches the given design, this means that spaces between elements, colors, typography, fonts, etc. should match.

cat

exp


design

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @AmineDerbal, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
✔️ Tests passing👏
👏 The app works as expected on the local machine.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AmineDerbal** [comment]
- Fixed the views to match the design 👍 
- Fixed the production error in the deployment 👍 
- On the deployed page once registered you will receive a confirmation email (check your spam folder in case) 👍 
- the new presentation video https://www.loom.com/share/af2e66d770d445999b3478945981f9a4?sid=03dd5e79-9193-46f4-8102-240b26c7d969 👍 
- for the deployment the email address should be valid 👍

**Review by AmaduKamara — APPROVED** [review]
# Status: Approved 🟢 🟢 🟢

Hi @AmineDerbal,

Good job working on the changes highlighted by the previous reviewer 👍🏽

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

## Highlights:

✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ The app works fine both locally and online 👍🏽
✅ Good video presentation 👍🏽
✅ Good commit messages 👍🏽
✅ Your README is professional 👍🏽

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**